app/src/lib/stores/app-store.ts

```diff
@@ -4736,19 +4752,33 @@ export class AppStore extends TypedBaseStore<IAppState> {
     await createDesktopStashEntry(repository, branchName)
   }
 
-  public async _popStash(repository: Repository, branch: Branch) {
-    const stash = await getLastDesktopStashEntryForBranch(
-      repository,
-      branch.name
-    )
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _popStashEntry(repository: Repository, stashEntry: IStashEntry) {
+    const gitStore = this.gitStoreCache.get(repository)
+    await gitStore.performFailableOperation(() => {
+      return popStashEntry(repository, stashEntry.stashSha)
+    })
+    log.info(`Popped stash with commit id ${stashEntry.stashSha}`)
 
-    if (stash === null) {
-      log.warn('no stash found that matches')
-      return
-    }
+    await this._refreshRepository(repository)
 
-    await popStashEntry(repository, stash.stashSha)
-    log.info(`Popped stash with commit id ${stash.stashSha}`)
+    this.emitUpdate()
+  }
+
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _dropStashEntry(
+    repository: Repository,
+    stashEntry: IStashEntry
+  ) {
+    const gitStore = this.gitStoreCache.get(repository)
+    await gitStore.performFailableOperation(() => {
+      return dropDesktopStashEntry(repository, stashEntry.stashSha)
+    })
+    log.info(`Dropped stash with commit id ${stashEntry.stashSha}`)
+
+    await this._refreshRepository(repository)
+
+    this.emitUpdate()
```

Comment:
Does this `emitUpdate` need to happen? I believe `_refreshRepository()` should cover this...

Reply:
Looks like we do need it.

https://github.com/desktop/desktop/blob/974b0a4790d46e82a4f2e99ffe2d7bdef8efd596/app/src/lib/stores/app-store.ts#L2206-L2259

My question is should we be emitting an update when we refresh the repo?